lib/lib.d.ts

```diff
@@ -1258,6 +1258,12 @@ interface Array<T> {
       * @param callbackfn A function that accepts up to three arguments. The filter method calls the callbackfn function one time for each element in the array.
       * @param thisArg An object to which the this keyword can refer in the callbackfn function. If thisArg is omitted, undefined is used as the this value.
       */
+    filter<S extends T>(callbackfn: (value: T, index: number, array: T[]) => value is S, thisArg?: any): S[];
```

Comment:
The files under `lib` are not edited manually, they are generated as part of the build. please remove them. you only need to update `src\lib`